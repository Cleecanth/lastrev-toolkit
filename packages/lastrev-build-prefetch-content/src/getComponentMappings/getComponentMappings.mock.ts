export const contentTypes = [
  { sys: { id: 'pageGeneral' } },
  { sys: { id: 'settingsGlobal' } },
  { sys: { id: 'moduleCardGeneral' } },
  { sys: { id: 'moduleHero' } }
];

export const componentNames = ['PageGeneral', 'Layout', 'ModuleCardGeneral', 'GlobalHeader'];
