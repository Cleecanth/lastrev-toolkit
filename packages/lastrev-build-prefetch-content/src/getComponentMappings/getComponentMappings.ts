import { ContentType } from 'contentful';
import { filter, each, includes, has, get } from 'lodash';
import pascalCase from '../helpers/pascalCase';

type MappingConfig = {
  overrides?: Record<string, string>;
  exclude?: string[];
};

const getComponentMappings = (contentTypes: ContentType[], config: MappingConfig): Record<string, string> => {
  const out = {};

  const overrides = get(config, 'overrides', {});
  const exclude = get(config, 'exclude', []);

  each(contentTypes, (item) => {
    const {
      sys: { id: contentTypeId }
    } = item;

    const componentName = get(overrides, contentTypeId, pascalCase(contentTypeId));

    if (!includes(exclude, componentName)) {
      out[contentTypeId] = componentName;
    }
  });

  return out;
};

export default getComponentMappings;
