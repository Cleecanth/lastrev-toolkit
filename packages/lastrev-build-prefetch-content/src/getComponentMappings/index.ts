import getComponentMappings from './getComponentMappings';

export default getComponentMappings;
