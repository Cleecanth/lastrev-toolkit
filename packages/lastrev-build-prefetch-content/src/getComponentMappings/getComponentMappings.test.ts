import { ContentType } from 'contentful';
import getComponentMappings from './getComponentMappings';
import { contentTypes } from './getComponentMappings.mock';

describe('getComponentMappings.js', () => {
  test('returns correct mappings with overrides and no exclude', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      overrides: {
        settingsGlobal: 'Layout'
      }
    });
    expect(mappings).toMatchSnapshot();
  });
  test('returns correct mappings with exclude and no overrides', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      exclude: ['PageGeneral']
    });
    expect(mappings).toMatchSnapshot();
  });
  test('returns correct mappings with both overrides and excludes', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      exclude: ['PageGeneral'],
      overrides: {
        settingsGlobal: 'Layout'
      }
    });
    expect(mappings).toMatchSnapshot();
  });
  test('returns correct mappings with override containing non-existent component', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      overrides: {
        pageGeneral: 'Dummy'
      }
    });
    expect(mappings).toMatchSnapshot();
  });
  test('returns correct mappings with override containing non-existent content type', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      overrides: {
        dummy: 'Dummy'
      }
    });
    expect(mappings).toMatchSnapshot();
  });
  test('returns correct mappings with exclude containing non-existent component', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      overrides: {
        dummy: 'Dummy'
      }
    });
    expect(mappings).toMatchSnapshot();
  });
  test('returns correct mappings with overrides but exclude contains overriden component', () => {
    const mappings = getComponentMappings(contentTypes as ContentType[], {
      exclude: ['Layout'],
      overrides: {
        settingsGlobal: 'Layout'
      }
    });
    expect(mappings).toMatchSnapshot();
  });
});
