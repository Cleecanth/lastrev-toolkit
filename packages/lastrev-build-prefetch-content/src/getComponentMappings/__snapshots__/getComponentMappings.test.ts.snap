// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentMappings.js returns correct mappings with both overrides and excludes 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "settingsGlobal": "Layout",
}
`;

exports[`getComponentMappings.js returns correct mappings with exclude and no overrides 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "settingsGlobal": "SettingsGlobal",
}
`;

exports[`getComponentMappings.js returns correct mappings with exclude containing non-existent component 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "pageGeneral": "PageGeneral",
  "settingsGlobal": "SettingsGlobal",
}
`;

exports[`getComponentMappings.js returns correct mappings with override containing non-existent component 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "pageGeneral": "Dummy",
  "settingsGlobal": "SettingsGlobal",
}
`;

exports[`getComponentMappings.js returns correct mappings with override containing non-existent content type 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "pageGeneral": "PageGeneral",
  "settingsGlobal": "SettingsGlobal",
}
`;

exports[`getComponentMappings.js returns correct mappings with overrides and no exclude 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "pageGeneral": "PageGeneral",
  "settingsGlobal": "Layout",
}
`;

exports[`getComponentMappings.js returns correct mappings with overrides but exclude contains overriden component 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "moduleHero": "ModuleHero",
  "pageGeneral": "PageGeneral",
}
`;
